Lib/timeit.py

```diff
@@ -328,6 +335,12 @@ def callback(number, time_taken):
 
         if verbose:
             print()
+    else:
+        if number % duplicate:
+            print("number is not divisible by duplicate.",
+                  file=sys.stderr)
+            return 2
+        number //= duplicate
```

Comment:
`--repeat` is not related here. Perhaps you meant `--number`.

`--number 10` runs the statement 10 times, in 10 iterations, 1 time per iteration. The total time is divided by 10.

`--number 10 --duplicate 2` runs the statement 10 times, in 5 iterations, 2 time per iteration. The total time is divided by 10.

If the overhead of iterating is insignificant, both results are the same. Due to the overhead of iterating the second result is smaller.

At line 309 the original statement is multiplied by `duplicate`, therefore we should divide the number of iterating by `duplicate`. The total number of running the original statement is left the same.